libs/cart/src/drivers/magento/cart.service.ts

```diff
@@ -57,15 +57,26 @@ export class DaffMagentoCartService implements DaffCartServiceInterface<DaffCart
       switchMap(items =>
 				// make the delete requests in parallel and collect them into a single observable
 				// return null if there are no items in the cart
-        items.length ? zip(...items.map(item =>
+        items.length ? forkJoin(...items.map(item =>
           this.cartItemDriver.delete(cartId, item.item_id)
         )) : of(null)
       ),
-      // find the most up to date delete response by looking for one with no items
-      // make a get request if there were no items in the cart
+			// make a get request if there were no items in the cart
+			// todo: find a better way to do this. Because all of the remove item requests are made
+			// asynchronously, there is no guarantee that any cart response will be the "last" one, i.e.
+			// the one with no more items.
 			switchMap(updatedCarts => updatedCarts ? 
```

Comment:
We can remove the condition here.

Reply:
We still need it in case you try to clear an empty cart. This was a bug from before